libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -0,0 +1,50 @@
+package add_host_metadata
+
+import (
+	sysinfo "github.com/elastic/go-sysinfo"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+	"github.com/elastic/go-sysinfo/types"
+)
+
+func init() {
+	processors.RegisterPlugin("add_host_metadata", newHostMetadataProcessor)
+}
+
+type addHostMetadata struct {
+	info types.HostInfo
+}
+
+func newHostMetadataProcessor(_ *common.Config) (processors.Processor, error) {
+	h, err := sysinfo.Host()
+	if err != nil {
+		return nil, err
+	}
+	return &addHostMetadata{
+		info: h.Info(),
+	}, nil
+}
+
+func (p addHostMetadata) Run(event *beat.Event) (*beat.Event, error) {
+
+	data := common.MapStr{
+		"host": common.MapStr{
+			"hostname": p.info.Hostname,
+			"id":       p.info.UniqueID,
+			"os": common.MapStr{
+				"architecture": p.info.Architecture,
```

Comment:
@andrewkroh Seeing that you have `Architecture` outside of `info` I have second thoughts on puttig it under `os` as I agree it more belongs to the `host` then the os itself.

Also thinking about putting `os` info on the top level.